addons/website/static/src/js/editor/wysiwyg.js

```diff
@@ -164,8 +164,22 @@ Wysiwyg.include({
     _saveElement: async function ($el, context, withLang) {
         var promises = [];
 
-        // Saving a view content
-        await this._super.apply(this, arguments);
+        // Saving Embed Code snippets with <script> in the database, as these
+        // elements are removed in edit mode.
+        if ($el[0].querySelector(".s_embed_code")) {
+            // Copied so as not to impact the actual DOM and prevent scripts
+            // from loading.
+            const clonedEl = $el[0].cloneNode(true);
+            for (const embeddedCodeEl of clonedEl.querySelectorAll(".s_embed_code_embedded")) {
+                if (embeddedCodeEl.dataset.embedContent) {
+                    embeddedCodeEl.innerHTML = embeddedCodeEl.dataset.embedContent;
+                }
+            }
+            await this._super.apply(this, [$(clonedEl), ...[...arguments].slice(1)]);
```

Comment:
Probably prettier to use the new syntax instead of apply:
```suggestion
            await this._super($(clonedEl), ...[...arguments].slice(1));
```

Also if possible to avoid `...[...arguments].slice(1)` it would be better. Since `context, withLang` are unused, you could name them `...rest` and use `...rest` here. Or you could leave them like that and add `...rest` anyway and use the three over here.
So
```javascript
async function (a, ...rest) {
    // ...
    await this._super(otherA, ...rest);
    // ...
}
```
Or
```javascript
async function (a, unusedB, unusedC, ...rest) {
    // ...
    await this._super(otherA, unusedB, unusedC, ...rest);
    // ...
}
```
I probably prefer the second one (always "declaring" all possible parameters in overrides) but both are fine.